src/PrestaShopBundle/Bridge/Helper/Listing/HelperBridge/HelperListBridge.php

```diff
@@ -157,42 +156,43 @@ protected function generateListQuery(
             'fields' => &$helperListConfiguration->fieldsList,
         ]);
 
-        if (!Validate::isTableOrIdentifier($helperListConfiguration->table)) {
-            throw new PrestaShopException(sprintf('Table name %s is invalid:', $helperListConfiguration->table));
+        if (!Validate::isTableOrIdentifier($helperListConfiguration->getTableName())) {
+            throw new PrestaShopException(sprintf('Table name %s is invalid:', $helperListConfiguration->getTableName()));
         }
 
         $limit = $this->checkSqlLimit($helperListConfiguration);
 
         /* Determine offset from current page */
         $start = 0;
-        if ((int) Tools::getValue('submitFilter' . $helperListConfiguration->listId)) {
-            $start = ((int) Tools::getValue('submitFilter' . $helperListConfiguration->listId) - 1) * $limit;
+        $page = $request->request->getInt('submitFilter' . $helperListConfiguration->getListId());
```

Comment:
It's a good thing you try to get rid of the usage of `Tools::getValue` but is it really needed? Since we said we tried to avoid changing the legacy code too much. Unless there's a bug when you use `Tools::getValue`?

Reply:
we are migrating controller, so i think we can at least try to use sf Request where possible. (but with Configuration values i would say its not worth for sure). Tools::getValue gets values from anywhere (no matter if its from post request or qyuery btw, so its not as precise, may raise bugs :man_shrugging: )

Reply:
That's why this improvement may raise some bugs, although you could use `$request->get` which does the same thing (uses GET or POST parameters). But even then I'm a bit worried 😅 We also have a backward conversion of Symfony parameters to legacy parameters via `Tools::setFallbackParameters` based on `_legacy_parameters`, even if I'm not sure it will be used in this case (it's more useful for vertical migrations)